### ISSUE TITLE
adding rows to grouped table are always added at the end and not in the right group.

### ISSUE THREAD

**Issue by shayo** [olifolkerd/tabulator#644]
I have a table that is grouped by a given column.
When I manually add rows to the table after its creation, they are always added at the end, and not to the group to which they belong.

calling  $("#table").tabulator("redraw");
doesn't rearrange the items.
Solution?

**Comment by shayo** [comment]
Is this somehow related to "groupStartOpen" ?
I tried adding groupStartOpen:true,but that didn't solve the problem.

**Comment by olifolkerd** [comment]
could you post an example of the ***addrow** function you are using and your table constructor

**Comment by shayo** [comment]
```

//Build Tabulator
$("#grouped_table").tabulator({
    height:"311px",
    groupBy:'label',  
    groupStartOpen:true,   
    responsiveLayout:true,
       rowClick:function(e, id, data, row){
        if ($.inArray(id,grouped_selected_rows) == -1)
       {
        grouped_selected_rows.push(id);
        grouped_selected_route_real_ids.push(data.real_id);
       } else 
       {
          var index = grouped_selected_rows.indexOf(id);
          if (index > -1) {
             grouped_selected_rows.splice(index, 1);
             grouped_selected_route_real_ids.splice(index, 1);
          }
       }
           },
      sortable:true,
    selectable:true, //make rows selectable
    columns:[
          {title:"Setup Date", field:"setup_date", align:"center",width:150},

      {title:"Id", field:"id", width:100},
             {title:"real_id", field:"real_id", align:"center",width:60,visible:false},

       {title:"Setter", field:"setter", align:"center",width:200},
{title:"Tape", field:"tape", align:"center",width:80

, formatter:function(value, data, cell, row, options, formatterParams){
        //value - the value of the cell
        //data - the data for the row the cell is in
        //cell - the DOM element of the cell
        //row - the DOM element of the row
        //options - the options set for this tabulator
        //formatterParams - parameters set for the column
        return "<div> <img src='"+data.tape+"' height='20px'> </div>"; // must return the html or jquery element of the html for the contents of the cell;
    }},



      {title:"Type", field:"type", align:"center",width:120},
      {title:"Grade", field:"grade", align:"center",width:80},
      {title:"Location", field:"location", align:"center",width:120},
      {title:"Anchor", field:"anchor", align:"center",width:120},
      {title:"Label", field:"label", align:"center",width:120},
       ],
    
});
```

```

function buildentry(k, label)
{
   table_entry = { real_id:routes[k].ID, id: routes[k].ID_SHORT, type: routes[k].TYPE, grade: routes[k].GRADE,setter: routes[k].firstname+" " + routes[k].lastname,
      setup_date:routes[k].DATE_ESTABLISHED.substring(0,10),  tape:routes[k].COLOR ,location:routes[k].wall_name,
      anchor:routes[k].ANCHOR, label:label};
      return table_entry;
}
 
```

Add row:
`
$("#grouped_table").tabulator("addRow", buildentry(k, label));`

**Comment by olifolkerd** [comment]
Your ***buildentry*** entry isnt returning anything.

You need to add `return table_entry;` at the end of the function

Cheers

Oli

**Comment by shayo** [comment]
eh? buildentry returns table_entry.

```

function buildentry(k, label)
{
   table_entry = { real_id:routes[k].ID, id: routes[k].ID_SHORT, type: routes[k].TYPE, grade: routes[k].GRADE,setter: routes[k].firstname+" " + routes[k].lastname,
      setup_date:routes[k].DATE_ESTABLISHED.substring(0,10),  tape:routes[k].COLOR ,location:routes[k].wall_name,
      anchor:routes[k].ANCHOR, label:label};
      return table_entry;
}
 
```

**Comment by olifolkerd** [comment]
Hey @shayo,

on further investigation Tabulator could definitily be handling this better, i will look at enhancing this functionality in the next release.

Cheers

Oli

**Comment by olifolkerd** [comment]
Hey @shayo 

You will be happy to hear i have pushed a fix for this to the 3.4 branch, it will be included in this evenings release.

Cheers

Oli :)

**Comment by olifolkerd** [comment]
Tabulator version 3.4 has now been officially released!

You can find details of the new release in the [News Section](http://tabulator.info/news)

Full documentation of all the new features can be found [Here](http://tabulator.info/docs/3.4)

Cheers

Oli :)
